### PR TITLE
Docs: Mkdocs Material Overhaul

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
Switches to MkDocs Material rather than sphinx + read the docs theme.

Builds here: http://dgasmith.github.io/opt_einsum/

## Status
- [ ] Ready to go

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+has been passed upstream and most of the same features that can be found in
+this repository can be enabled with ``numpy.einsum(..., optimize=True)``. However,
+this repository often has more up to date algorithms for complex contractions.
+Several advanced features are as follows:
+
+* Inspect [detailed information](paths/introduction.md) about the path chosen.
+* Perform contractions with [numerous backends](getting_started/backends.md), including on the GPU and with libraries such as [TensorFlow](https://www.tensorflow.org) and [PyTorch](https://pytorch.org).
+* Generate [reusable expressions](getting_started/reusing_paths.md), potentially with constant tensors, that can be compiled for greater performance.
+* Use an arbitrary number of indices to find contractions for [hundreds or even thousands of tensors](examples/large_expr_with_greedy.md).
+* Share `intermediate computations <sharing_intermediates.html) among multiple contractions.
```
Comment (dgasmith) [inline docs/index.md:23]:
```suggestion
* Share [intermediate computations](./sharing_intermediates.md) among multiple contractions.
```

File: docs/changelog.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
+Changelog
+=========
+
+## 3.3.0 / 2020-07-19
+
+Adds a `object` backend for optimized contractions on arbitrary Python objects.
+
+**New Features**
+
+ - [\#145](https://github.com/EntosAI/codex/pull/145) Adds a `object` based backend so that `contract(backend='object')` can be used on arbitrary objects such as SymPy symbols.
```
Comment (jcmgray) [inline docs/changelog.md:10]: I think all these references to `EntosAI/codex` should be ``dgasmith/opt_einsum`` right?

File: docs/changelog.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+time resulting in non-optimal scaling for contractions with many terms.
+Opt_einsum aims to fix this by optimizing the contraction order which can lead
+to arbitrarily large speed ups at the cost of additional intermediate tensors.
+
+Opt_einsum is also implemented into the np.einsum function as of NumPy v1.12.
+
+**New Features**
+
+- Tensor contraction order optimizer.
+- :func:`opt_einsum.contract` as a drop-in replacement for :func:`numpy.einsum`.
```
Comment (jcmgray) [inline docs/changelog.md:187]: These style of references don't seem to work in the new docs.

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+has been passed upstream and most of the same features that can be found in
+this repository can be enabled with ``numpy.einsum(..., optimize=True)``. However,
+this repository often has more up to date algorithms for complex contractions.
+Several advanced features are as follows:
+
+* Inspect [detailed information](paths/introduction.md) about the path chosen.
+* Perform contractions with [numerous backends](getting_started/backends.md), including on the GPU and with libraries such as [TensorFlow](https://www.tensorflow.org) and [PyTorch](https://pytorch.org).
+* Generate [reusable expressions](getting_started/reusing_paths.md), potentially with constant tensors, that can be compiled for greater performance.
+* Use an arbitrary number of indices to find contractions for [hundreds or even thousands of tensors](examples/large_expr_with_greedy.md).
+* Share [intermediate computations](./sharing_intermediates.md) among multiple contractions.
```
Comment (jcmgray) [inline docs/index.md:23]: This link also seems to be broken, needs to be ``getting_started/sharing_intermediates.md`` maybe?

File: mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+nav:
+  - Overview: index.md
+  - Getting Started:
+    - Installing: getting_started/install.md
+    - Input Format: getting_started/input_format.md
+    - "Backends & GPU Support": getting_started/backends.md
+    - Reusing Paths: getting_started/reusing_paths.md
+    - Sharing Intermediates: getting_started/sharing_intermediates.md
+  - Path Infomration:
```
Comment (jcmgray) [inline mkdocs.yml:42]: small typo here

**Review by jcmgray — APPROVED** [review]
New docs look good! Mostly just spotted a couple of places links seem broken.

File: docs/changelog.md (hunk 3 above)
Comment (dgasmith) [reply docs/changelog.md:187]: It is a fair amount of work to overhaul, (a drawback of the current mkdocs, I didn't find a nice replacement for the above). I think we can simply remove the `:func:` and leave it for now?

**Comment by dgasmith** [comment]
Thank you for the careful review @jcmgray, quite a few lines changes. Going ahead and merging this in.

**Pull request merged**